infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -710,15 +707,19 @@ private void handleAllocation(RequestMsg req, ChannelHandlerContext ctx, IServer
 
         // update the cache of conflict parameters
         if (tokenRequest.hasTxnResolution()) {
-            tokenRequest.getTxnResolution().getWriteConflictParamsSetList()
-                    .forEach((item) -> {
-                        // insert an entry with the new timestamp using the
-                        // hash code based on the param and the stream id.
-                        item.getValueList().forEach(conflictParam ->
-                                cache.put(new ConflictTxStream(getUUID(item.getKey()),
-                                        conflictParam.toByteArray(), newTail - 1)));
-                    });
+            Set<ConflictTxStream> conflictTxStreamList = tokenRequest.getTxnResolution()
+                    .getWriteConflictParamsSetList()
+                    .stream()
+                    .flatMap(item -> {
+                        final UUID streamId = getUUID(item.getKey());
```

Comment:
You don't technically need this variable.

Reply:
I used it to avoid creating duplicate UUID objects for the same stream id.